tools/templates/vfvs_run.py

```diff
@@ -216,32 +222,159 @@ def program_runstring_array(task):
+        cmd = ['{}/PLANTS'.format(task['tools_path']), 
+               '--mode', 'screen', 
+                '{}.txt'.format(str(uuid.uuid4()))]
+    elif(task['program'] == "AutodockZN"):
+        with open(task['config_path']) as fd:
+            config_ = dict(read_config_line(line) for line in fd)
+        for item in config_: 
+            if '#' in config_[item]: config_[item] = config_[item].split('#')[0]
+            else: continue 
+        cmd = ['{}/AutodockVina_1.2'.format(task['tools_path']), 
+               '--ligand', '{}'.format(task['ligand_path']), 
+               '--maps', config_['afinit_maps_name'], 
+               '--scoring', 'ad4', 
+               '--exhaustiveness', '{}'.format(config_['exhaustiveness']), 
+               '--out', '{}'.format(task['output_path'])]
+    elif(task['program'] == "gnina"):
+        with open(task['config_path']) as fd:
+            config_ = dict(read_config_line(line) for line in fd)
+        for item in config_: 
+            if '#' in config_[item]: config_[item] = config_[item].split('#')[0]
+            else: continue 
+        cmd = ['{}/gnina'.format(task['tools_path']), 
+               '-r', config_['receptor'], 
+               '-l', '{}'.format(task['ligand_path']), 
+               '--exhaustiveness', '{}'.format(config_['exhaustiveness']), 
+               '--center_x', '{}'.format(config_['center_x']),
+               '--center_y', '{}'.format(config_['center_y']),
+               '--center_z', '{}'.format(config_['center_z']),
+               '--size_x',   '{}'.format(config_['size_x']),
+               '--size_y',   '{}'.format(config_['size_y']),
+               '--size_z',   '{}'.format(config_['size_z']),  
+               '--out', '{}'.format(task['output_path'])]
+    elif(task['program'] == "rDock"):
+        
+        with open(task['config_path']) as fd:
+            config_ = dict(read_config_line(line) for line in fd)
+        for item in config_: 
+            if '#' in config_[item]: config_[item] = config_[item].split('#')[0]
+            else: continue 
+        cmd = [ 'rbdock', 
+                '-i', task['ligand_path'], 
+                '-o', task['output_path'],
+                '-r', config_['rdock_config'], 
+                '-p', config_['dock_prm'],
+                '-n', config_['runs']]
+        
+    elif(task['program'] == "M-Dock"):
+        
+        with open(task['config_path']) as fd:
+            config_ = dict(read_config_line(line) for line in fd)
+        for item in config_: 
+            if '#' in config_[item]: config_[item] = config_[item].split('#')[0]
+        cmd = ['{}/MDock_Linux'.format(task['tools_path']), 
+                config_['protein_name'], 
+                task['ligand_path'],
+               '-param', config_['mdock_config']
+              ]
+    elif(task['program'] == "MCDock"):
+    
+        cmd = ['{}/mcdock'.format(task['tools_path']), 
+                '--target', config_['protein_name'], 
```

Comment:
where does this come from?

Reply:
A user has to compile and place the executable (named 'mcdock') within the tools_path; this is mentioned in "Running with MCDock" inside DOCKING_SOFTWARE_NOTES.md